openedx/core/djangoapps/user_authn/views/login.py

```diff
@@ -401,6 +401,8 @@ def login_user(request):
 
     set_custom_metric('login_user_course_id', request.POST.get('course_id'))
 
+    if settings.FEATURES.get('ENABLE_THIRD_PARTY_ONLY_AUTH') and not third_party_auth_requested:
+        return HttpResponseForbidden("First party login not supported")
```

Comment:
I am not aware of "First party login" being an industry-accepted term.

Can we rephrase the exception message to something like: "Third party authentication is required to login. Username and password were received instead."